sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -263,11 +295,21 @@ case class Invoke(
     propagateNull: Boolean = true,
     returnNullable : Boolean = true) extends InvokeLike {
 
+  lazy val argClasses = ScalaReflection.expressionJavaClasses(arguments)
+
   override def nullable: Boolean = targetObject.nullable || needNullCheck || returnNullable
   override def children: Seq[Expression] = targetObject +: arguments
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  override def eval(input: InternalRow): Any = {
+    val obj = targetObject.eval(input)
+    if (obj == null) {
+      // return null if obj is null
+      null
+    } else {
+      val method = obj.getClass.getDeclaredMethod(functionName, argClasses : _*)
```

Comment:
Why are we resolving during evaluation time? This seems to be expensive?

Reply:
Unlike `StaticInvoke`, this is because `obj` is dynamically evaluated with a value in `InternalRow`. We cannot resolve beforehand when the case class is created.

Reply:
We should know the type before this right? Otherwise whole stage code generation doesn't work either.

Reply:
We can know the base class beforehand. The `obj` class may be a subclass. I think this is a kind of virtual table lookup based on the literal name?
Since JVM class resolution converts the literal name to a constant offset, the code generation can perform a fast lookup based on the offset.

Reply:
Ah, this method may be applied to the base class. let me try this optimization.

Reply:
Good catch. I also use `method` that was already defined for interpreted execution.